web/src/message_list.js

```diff
@@ -110,14 +114,14 @@ export class MessageList {
             render_info = this.append_to_view(bottom_messages, opts);
         }
 
-        if (this.narrowed && !this.visibly_empty()) {
+        if (!this.visibly_empty()) {
             // If adding some new messages to the message tables caused
             // our current narrow to no longer be empty, hide the empty
             // feed placeholder text.
             narrow_banner.hide_empty_narrow_message();
         }
 
-        if (this.narrowed && !this.visibly_empty() && this.selected_id() === -1) {
+        if (!this.visibly_empty() && this.selected_id() === -1) {
             // The message list was previously empty, but now isn't
             // due to adding these messages, and we need to select a
             // message. Regardless of whether the messages are new or
```

Comment:
I think these checks might still need a conditional on whether the view is currently rendered instead of `this.narrowed`; if it isn't, then selecting a message or hiding the banner will be applied based on the properties of an invisible background view, and that seems buggy? So maybe we need `this === message_lists.current`, effectively?

Reply:
It might make sense to just have an explicit `this.is_current_message_list` boolean on this object that we can check here; it'd likely be easy for `narrow.ts` logic to maintain that, and would avoid needing a messy import of some sort of access `message_lists.current`.

Reply:
Pushed a commit for this in #29871.